lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -234,6 +244,39 @@ monitorStakePools tr gp nl db@DBLayer{..} = do
                 putPoolRegistration slot pool
         pure Continue
 
+monitorMetadata
+    :: Tracer IO StakePoolLog
+    -> GenesisParameters
+    -> (StakePoolMetadataUrl -> StakePoolMetadataHash -> IO (Either String StakePoolMetadata))
+    -> DBLayer IO
+    -> IO ()
+monitorMetadata tr gp fetchMetadata DBLayer{..} = forever $ do
+    refs <- atomically unfetchedPoolMetadataRefs
+
+    successes <- fmap catMaybes $ forM refs $ \(pid, url, hash) -> do
+        traceWith tr $ MsgFetchPoolMetadata pid url
+        fetchMetadata url hash >>= \case
+            Left msg -> Nothing <$ do
+                traceWith tr $ MsgFetchPoolMetadataFailure pid msg
```

Comment:
Wouldn't we want to store the failure in the db to avoid the worker retrying bad links forever?

Reply:
> Wouldn't we want to store the failure in the db to avoid the worker retrying bad links forever?

For now, failures are just retried every next block. I'll do something a little smarter in another PR. My plan is to keep **in-memory** some indexes with the failing ones to avoid retrying them more than X every hour. I don't think it's necessary to store anything in the database here, if the wallet is restarted, it'll retry right away and then wait again until the next hour before re-retrying. I see it as a nice feature :)